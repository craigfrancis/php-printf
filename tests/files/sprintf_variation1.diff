--
     string(3) "0.5"
     
     -- Iteration 10 --
049- sprintf(): Argument #1 ($format) must be of type string, array given
050- sprintf(): Argument #1 ($format) must be of type string, array given
051- sprintf(): Argument #1 ($format) must be of type string, array given
049+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 71
050+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 78
051+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 85
     
     -- Iteration 11 --
054- sprintf(): Argument #1 ($format) must be of type string, array given
055- sprintf(): Argument #1 ($format) must be of type string, array given
056- sprintf(): Argument #1 ($format) must be of type string, array given
054+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 71
055+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 78
056+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 85
     
     -- Iteration 12 --
059- sprintf(): Argument #1 ($format) must be of type string, array given
060- sprintf(): Argument #1 ($format) must be of type string, array given
061- sprintf(): Argument #1 ($format) must be of type string, array given
059+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 71
060+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 78
061+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 85
     
     -- Iteration 13 --
064- sprintf(): Argument #1 ($format) must be of type string, array given
065- sprintf(): Argument #1 ($format) must be of type string, array given
066- sprintf(): Argument #1 ($format) must be of type string, array given
064+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 71
065+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 78
066+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 85
     
     -- Iteration 14 --
069- sprintf(): Argument #1 ($format) must be of type string, array given
070- sprintf(): Argument #1 ($format) must be of type string, array given
071- sprintf(): Argument #1 ($format) must be of type string, array given
069+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 71
070+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 78
071+ user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 85
     
     -- Iteration 15 --
     string(1) "1"
--
     string(6) "Object"
     
     -- Iteration 22 --
109- sprintf(): Argument #1 ($format) must be of type string, resource given
110- sprintf(): Argument #1 ($format) must be of type string, resource given
111- sprintf(): Argument #1 ($format) must be of type string, resource given
109+ user_sprintf(): Argument #1 ($format) must be of type string, resource given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 71
110+ user_sprintf(): Argument #1 ($format) must be of type string, resource given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 78
111+ user_sprintf(): Argument #1 ($format) must be of type string, resource given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 85
     Done
