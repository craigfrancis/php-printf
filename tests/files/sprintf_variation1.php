<?php
/*
* Testing user_sprintf() : with different unexpected values for format argument other than the strings
*/

echo "*** Testing sprintf() : with unexpected values for format argument ***\n";

// initialing required variables
$arg1 = "second arg";
$arg2 = "third arg";

// declaring class
class sample
{
  public function __toString() {
    return "Object";
  }
}

// creating a file resource
$file_handle = fopen(__FILE__, 'r');

//array of values to iterate over
$values = array(

      // int data
      0,
      1,
      12345,
      -2345,

      // float data
      10.5,
      -10.5,
      10.1234567e10,
      10.7654321E-10,
      .5,

      // array data
      array(),
      array(0),
      array(1),
      array(1, 2),
      array('color' => 'red', 'item' => 'pen'),

      // boolean data
      true,
      false,
      TRUE,
      FALSE,

      // empty data
      "",
      '',

      // object data
      new sample(),

      // resource data
      $file_handle
);

// loop through each element of the array for format

$count = 1;
foreach($values as $value) {
  echo "\n-- Iteration $count --\n";

  // with default argument
  try {
    var_dump(user_sprintf($value));
  } catch (TypeError $exception) {
    echo $exception->getMessage() . "\n";
  }

  // with two arguments
  try {
    var_dump(user_sprintf($value, $arg1));
  } catch (TypeError $exception) {
    echo $exception->getMessage() . "\n";
  }

  // with three arguments
  try {
    var_dump(user_sprintf($value, $arg1, $arg2));
  } catch (TypeError $exception) {
    echo $exception->getMessage() . "\n";
  }

  $count++;
}

// close the resource
fclose($file_handle);

echo "Done";
?>
