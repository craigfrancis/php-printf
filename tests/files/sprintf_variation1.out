*** Testing sprintf() : with unexpected values for format argument ***

-- Iteration 1 --
string(1) "0"
string(1) "0"
string(1) "0"

-- Iteration 2 --
string(1) "1"
string(1) "1"
string(1) "1"

-- Iteration 3 --
string(5) "12345"
string(5) "12345"
string(5) "12345"

-- Iteration 4 --
string(5) "-2345"
string(5) "-2345"
string(5) "-2345"

-- Iteration 5 --
string(4) "10.5"
string(4) "10.5"
string(4) "10.5"

-- Iteration 6 --
string(5) "-10.5"
string(5) "-10.5"
string(5) "-10.5"

-- Iteration 7 --
string(12) "101234567000"
string(12) "101234567000"
string(12) "101234567000"

-- Iteration 8 --
string(13) "1.07654321E-9"
string(13) "1.07654321E-9"
string(13) "1.07654321E-9"

-- Iteration 9 --
string(3) "0.5"
string(3) "0.5"
string(3) "0.5"

-- Iteration 10 --
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 71
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 78
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 85

-- Iteration 11 --
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 71
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 78
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 85

-- Iteration 12 --
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 71
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 78
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 85

-- Iteration 13 --
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 71
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 78
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 85

-- Iteration 14 --
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 71
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 78
user_sprintf(): Argument #1 ($format) must be of type string, array given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 85

-- Iteration 15 --
string(1) "1"
string(1) "1"
string(1) "1"

-- Iteration 16 --
string(0) ""
string(0) ""
string(0) ""

-- Iteration 17 --
string(1) "1"
string(1) "1"
string(1) "1"

-- Iteration 18 --
string(0) ""
string(0) ""
string(0) ""

-- Iteration 19 --
string(0) ""
string(0) ""
string(0) ""

-- Iteration 20 --
string(0) ""
string(0) ""
string(0) ""

-- Iteration 21 --
string(6) "Object"
string(6) "Object"
string(6) "Object"

-- Iteration 22 --
user_sprintf(): Argument #1 ($format) must be of type string, resource given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 71
user_sprintf(): Argument #1 ($format) must be of type string, resource given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 78
user_sprintf(): Argument #1 ($format) must be of type string, resource given, called in /Volumes/WebServer/Projects/php.fprint/tests/files/sprintf_variation1.php on line 85
Done